### ISSUE TITLE
Linter does not recognize same docstring issues as flake8 on command line

### ISSUE THREAD

**Issue by tdheff** [AtomLinter/linter-flake8#116]
On Atom version 1.0.5

The linter always raises "D100 - Missing docstring in public module" even if said docstring is present, but does not raise any other valid docstring warnings.

Example:
![image](https://cloud.githubusercontent.com/assets/1824571/11402600/bab00400-9366-11e5-83a0-a7e75fb2e45c.png)

vs. 

```
ff/team/__init__.py:5:1: D101 Missing docstring in public class
ff/team/__init__.py:12:1: D102 Missing docstring in public method
ff/team/__init__.py:18:1: D105 Missing docstring in magic method
```

**Comment by Arcanemagus** [comment]
This just forwards on the messages from `flake8`, are you sure Atom is using the same `flake8` you are testing with? (Different virtualenv?)

**Comment by dragonxlwang** [comment]
I am also spotting this problem all the time. I have specified the path of flake8 but it seems the error still does not match what outputted from flake8 directly. For example,

![screen shot 2015-11-29 at 5 58 15 pm](https://cloud.githubusercontent.com/assets/15812510/11461135/cd8d84b2-96c2-11e5-8140-5e6b9ba84c7b.png)
![screen shot 2015-11-29 at 5 58 15 pm](https://cloud.githubusercontent.com/assets/15812510/11461136/dfac2dba-96c2-11e5-931b-6b2169da0f58.png)

**Comment by Arcanemagus** [comment]
Again, are you sure that the `flake8` running in Atom is the same one you are testing in the terminal?

**Comment by Arcanemagus** [comment]
Also, when testing from the command line make sure you are using `stdin` as the input:
```ShellSession
cat dep_w2v.py | flake8
```

**Comment by dragonxlwang** [comment]
yes... I only have one flake:

$ for p in ${(s.:.)PATH}; do echo $p; ll $p | grep flake8; done
/sw/bin
ls: /sw/bin: No such file or directory
/sw/sbin
ls: /sw/sbin: No such file or directory
/Library/Frameworks/Python.framework/Versions/2.7/bin
ls: /Library/Frameworks/Python.framework/Versions/2.7/bin: No such file or directory
/Library/Frameworks/Python.framework/Versions/3.3/bin
ls: /Library/Frameworks/Python.framework/Versions/3.3/bin: No such file or directory
/usr/local/bin
-rwxr-xr-x    1 xiaolong  admin   232B Nov 26 14:08 flake8
/usr/bin
/bin
/usr/sbin
/sbin
/opt/X11/bin
/Library/TeX/texbin
/Users/xiaolong/local
ls: /Users/xiaolong/local: No such file or directory
/Library/Frameworks/Python.framework/Versions/3.3/bin
ls: /Library/Frameworks/Python.framework/Versions/3.3/bin: No such file or directory
/Library/Frameworks/Python.framework/Versions/2.7/bin
ls: /Library/Frameworks/Python.framework/Versions/2.7/bin: No such file or directory
/Users/xiaolong/local
ls: /Users/xiaolong/local: No such file or directory
/Library/Frameworks/Python.framework/Versions/3.3/bin
ls: /Library/Frameworks/Python.framework/Versions/3.3/bin: No such file or directory
/Library/Frameworks/Python.framework/Versions/2.7/bin
ls: /Library/Frameworks/Python.framework/Versions/2.7/bin: No such file or directory
/Users/xiaolong/local
ls: /Users/xiaolong/local: No such file or directory
/Library/Frameworks/Python.framework/Versions/3.3/bin
ls: /Library/Frameworks/Python.framework/Versions/3.3/bin: No such file or directory
/Library/Frameworks/Python.framework/Versions/2.7/bin
ls: /Library/Frameworks/Python.framework/Versions/2.7/bin: No such file or directory

**Comment by dragonxlwang** [comment]
I tried to pipe in the file but flake8 seems not work that way:

$ cat dep_w2v.py | flake8
Usage: flake8 [options] input ...

flake8: error: input not specified

**Comment by Arcanemagus** [comment]
Ah, sorry you need a `-` at the end: `cat dep_w2v.py | flake8 -`

**Comment by dragonxlwang** [comment]
Oh yeah, I figured that out. But it's wired... After piping the file to flake8, it seems that flake8 output indeed throws D100:
```
$ cat dep_w2v.py | flake8 -
stdin:1:1: D100 Missing docstring in public module
stdin:2:1: F401 'nltk' imported but unused
stdin:8:1: F401 'numpy' imported but unused
stdin:9:1: F401 'sys' imported but unused
stdin:124:5: F841 local variable 'W' is assigned to but never used
```

**Comment by Arcanemagus** [comment]
Which pep257 plugin are you using? `flake8-docstrings` or `flake8-pep257`?

**Comment by dragonxlwang** [comment]
```
$ flake8 --version
2.5.0 (pep8: 1.6.2, pyflakes: 1.0.0, flake8-docstrings: 0.2.4, pep257: 0.7.0, mccabe: 0.3.1) CPython 2.7.10 on Darwin
```

Seems flake8-docstring. This comes from flake8 I believe

**Comment by Arcanemagus** [comment]
Can you try downgrading `pep8` to 1.5.7? That is the latest currently released version that is compatible with `flake8-docstrings`.

(`Requirement already up-to-date: pep8!=1.6.0,!=1.6.1,!=1.6.2,>=1.5.7 in c:\python27\lib\site-packages (from flake8->flake8-docstrings)`)

**Comment by dragonxlwang** [comment]
sure, but sorry can you let me know how to downgrade pep8 in flake8?

**Comment by Arcanemagus** [comment]
`pip install pep8==1.5.7`

**Comment by dragonxlwang** [comment]
I can only downgrade flake8 as a whole to 2.3.0
'''
$ flake8 --version
2.3.0 (pep8: 1.5.7, pyflakes: 1.0.0, flake8-docstrings: 0.2.4, pep257: 0.7.0, mccabe: 0.3.1) CPython 2.7.10 on Darwin
'''
And using that I am still getting:

$ cat dep_w2v.py | flake8 -
stdin:1:1: D100 Missing docstring in public module
stdin:2:1: F401 'nltk' imported but unused
stdin:8:1: F401 'numpy' imported but unused
stdin:9:1: F401 'sys' imported but unused
stdin:124:5: F841 local variable 'W' is assigned to but never used

**Comment by Arcanemagus** [comment]
I'm pretty sure v2.3.0 of `flake8` doesn't support the latest pep257 :stuck_out_tongue:. You should be able to just do `pip install -U flake8-docstrings` and it should update that to the latest version again.

In any case this seems like an issue with `flake8-docstrings` so I'm marking this as closed. You should file an issue [over there](https://gitlab.com/pycqa/flake8-docstrings/issues) about this.

**Comment by Arcanemagus** [comment]
You can also try `flake8-pep257`, they _just_ released 1.0.5 which should fix `stdin` on there.

**Comment by dragonxlwang** [comment]
thanks! after uninstall flake8-docstrings and install flake8-pep257:
```
$ flake8 --version
2.5.0 (pep8: 1.5.7, pyflakes: 1.0.0, flake8-pep257: 1.0.5, mccabe: 0.3.1) CPython 2.7.10 on Darwin
```
I can confirm that the problem seems no longer present.

```
$ cat dep_w2v.py | flake8 -
stdin:2:1: F401 'nltk' imported but unused
stdin:8:1: F401 'numpy' imported but unused
stdin:9:1: F401 'sys' imported but unused
stdin:25:1: D103 Missing docstring in public function
stdin:38:1: D103 Missing docstring in public function
stdin:53:1: D103 Missing docstring in public function
stdin:62:1: D103 Missing docstring in public function
stdin:77:1: D103 Missing docstring in public function
stdin:98:1: D103 Missing docstring in public function
stdin:109:1: D103 Missing docstring in public function
stdin:124:5: F841 local variable 'W' is assigned to but never used
stdin:131:1: D103 Missing docstring in public function
```

**Comment by tdheff** [comment]
Great news, thanks @Arcanemagus!

**Comment by Arcanemagus** [comment]
FYI, looked into it a bit and re-opened my original issue over on `flake8-docstrings`: https://gitlab.com/pycqa/flake8-docstrings/issues/2#note_2807528

It's simply reporting a D100 error for any file passed in via `stdin`.

**Comment by jay0lu** [comment]
I installed flake8-pep257, and the warning gone.
`pip install flake8-pep257`
